home/views.py

```diff
@@ -293,28 +299,34 @@ def addLongRebateBill(request):
 
     :template:`home/longRebate.html`
 
-    Gets the data from the log term rebate form , and adds it to the coresponding rebate bill
-    This form can only be accessed by the Institute's admin
+    Gets the data from the long-term rebate form and adds it to the corresponding rebate bill.
+    This form can only be accessed by the Institute's admin.
     """
     text = ""
-    if request.method == "POST" and request.user.is_authenticated:
+    try:
+        student = Student.objects.get(email__iexact=request.user.email)
+        if not student.allocation_enabled:
+            text = "You are not eligible to apply for long rebate. For more details, please inquire at the Dining Office."
+    except Student.DoesNotExist:
+        text = "Email ID does not exist in the database. Please log in using the correct email ID."
+        return render(request, "longRebate.html", {"text": text})
+
+    if request.method == "POST":

```

Comment:
```suggestion
    if request.method == "POST" and request.user.is_authenticated:
```

Reply:
ohh I see you moved it https://github.com/DaemonLab/Mess-Website/pull/107/files#diff-e45ecc06f7114902d8b222e2408b9de067157baeec1cddc7f59bb7dad190552bR21
But we should handle this at backend too. someone can use curl or postman to send a request.